test/integration/shared-worker/shared-worker.test.ts

```diff
@@ -0,0 +1,740 @@
+      enableEventEngine: true,
+      autoNetworkDetection: false,
+    });
+  });
+
+  afterEach(() => {
+    pubnubWithWorker.removeAllListeners();
+    pubnubWithWorker.unsubscribeAll();
+    pubnubWithWorker.destroy(true);
+
+    pubnubWithoutWorker.removeAllListeners();
+    pubnubWithoutWorker.unsubscribeAll();
+    pubnubWithoutWorker.destroy(true);
+  });
+
+  describe('Subscription Functionality with shared worker', () => {
+    it('should successfully subscribe to channels with shared worker', (done) => {
+      const channel = testChannels[0];
+      let connectionEstablished = false;
+      let errorReceived = false;
+
+      pubnubWithWorker.addListener({
+        status: (statusEvent) => {
+          if (statusEvent.category === PubNub.CATEGORIES.PNConnectedCategory && !connectionEstablished) {
+            connectionEstablished = true;
+            try {
+              // Verify successful connection (error can be undefined or false for success)
+              expect(statusEvent.error).to.satisfy((error: any) => error === false || error === undefined);
+              if (Array.isArray(statusEvent.affectedChannels)) {
+                expect(statusEvent.affectedChannels).to.include(channel);
+              }
+              done();
+            } catch (error) {
+              done(error);
+            }
+          } else if (statusEvent.category === PubNub.CATEGORIES.PNNetworkIssuesCategory && !errorReceived) {
+            errorReceived = true;
+            console.error('Network/Worker Error:', {
+              category: statusEvent.category,
+              error: statusEvent.error,
+              operation: statusEvent.operation,
+            });
+            done(new Error(`Shared worker failed to initialize: ${statusEvent.error || 'Unknown error'}`));
+          } else if (statusEvent.error && !connectionEstablished && !errorReceived) {
+            errorReceived = true;
+            done(new Error(`Status error: ${statusEvent.error}`));
+          }
+        },
+      });
+
+      const subscription = pubnubWithWorker.channel(channel).subscription();
+      subscription.subscribe();
+    }).timeout(10000);
+
+    it('should handle subscription changes correctly', (done) => {
+      const channel1 = testChannels[0];
+      const channel2 = testChannels[1];
+      let subscriptionCompleted = false;
+
+      pubnubWithWorker.addListener({
+        status: (statusEvent) => {
+          if (statusEvent.category === PubNub.CATEGORIES.PNConnectedCategory && !subscriptionCompleted) {
+            subscriptionCompleted = true;
+
+            // Check if we have the channels we expect
+            const currentChannels = pubnubWithWorker.getSubscribedChannels();
+            console.log(`Connected channels: ${currentChannels.join(',')}`);
+
+            try {
+              // Verify we're connected to at least one channel
+              expect(currentChannels.length).to.be.greaterThan(0);
```

Comment:
Since it is an instance with new flow, then should we wait for subscription changed "twice" to be sure, that we subscribed for both?

Reply:
Also, there were issues on the `SharedWorker` side where it ignored subscribing to new channels and, as a result, it didn't receive updates from them until some message will be sent to the original set of channels.
Writing it because check of `getSubscribedChannels` to verify outcome with `SharedWorker` won't test it properly. If real keys are used, maybe verify it by sending a message to one of the new channels in the subscription set - which you have a bit later in this test suite.